armi/testing/singleMixedAssembly.py

```diff
@@ -0,0 +1,258 @@
+        coolant: *component_coolant
+        duct: *component_duct
+        intercoolant: *component_intercoolant
+        axial expansion target component: fuel
+
+    plenum 2pin: &block_plenum_multiPin
+        grid name: twoPin
+        gap: &component_plenummultiPin_gap1
+            shape: Circle
+            material: Void
+            Tinput: 25.0
+            Thot: 600.0
+            id: 0.0
+            od: clad.id
+            latticeIDs: [1]
+        clad: *component_fuelmultiPin_clad1
+        wire: *component_fuelmultiPin_wire1
+        gap test:
+            <<: *component_plenummultiPin_gap1
+            od: clad test.id
+            latticeIDs: [2]
+        clad test: *component_fuelmultiPin_clad2
+        coolant: *component_coolant
+        duct: *component_duct
+        intercoolant: *component_intercoolant
+        axial expansion target component: clad test
+
+    mixed fuel plenum 2pin: &block_mixed_multiPin
+        grid name: twoPin
+        gap: *component_plenummultiPin_gap1
+        clad: *component_fuelmultiPin_clad1
+        wire: *component_fuelmultiPin_wire1
+        fuel test: *component_fuelmultiPin_fuel2
+        bond test: *component_fuelmultiPin_bond2
+        clad test: *component_fuelmultiPin_clad2
+        coolant: *component_coolant
+        duct: *component_duct
+        intercoolant: *component_intercoolant
+        axial expansion target component: fuel test
+
+    aclp plenum 2pin: &block_aclp_multiPin
+        <<: *block_plenum_multiPin
+
+    SodiumBlock: &block_dummy
+        flags: dummy
+        coolant:
+            shape: Hexagon
+            material: Sodium
+            Tinput: 25.0
+            Thot: 450.0
+            ip: 0.0
+            mult: 1.0
+            op: 19.0
+"""
+
+REGULAR_ASSEMBLY_DEF = """
+assemblies:
+    multi pin fuel:
+        specifier: LA
+        blocks: [*block_grid_plate, *block_fuel_multiPin_axial_shield, *block_fuel_multiPin, *block_fuel_multiPin, *block_fuel_multiPin, *block_mixed_multiPin, *block_aclp_multiPin, *block_plenum_multiPin, *block_duct, *block_dummy]
+        height: [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]
+        axial mesh points: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
+        material modifications:
+            U235_wt_frac: ['', '', 0.2, 0.2, 0.2, 0.2, '', '', '', '']
+            ZR_wt_frac: ['', '', 0.07, 0.07, 0.07, 0.07, '', '', '', '']
+        xs types: [A, A, B, C, C, D, A, A, A, A]
+"""  # noqa: E501
+
+GRID_DEFINITION = """
+grids:
+    core:
+        geom: hex
+        symmetry: third periodic
+        lattice map: LA
+    twoPin:
+        geom: hex_corners_up
+        symmetry: full
+        lattice map: |
+          - - -  1 1 1 1
+            - - 1 1 2 1 1
+             - 1 1 2 2 1 1
+              1 1 2 2 2 1 1
+               1 1 2 2 1 1
+                1 1 2 1 1
+                 1 1 1 1
+"""
+
+
+def buildMixedPinAssembly():
+    FULL_BP = BLOCK_DEFINITIONS + REGULAR_ASSEMBLY_DEF + GRID_DEFINITION
```

Comment:
Can we put a docstring here that says this is a mixed-pin assembly for a fast-reactor hex-assembly? Or whatever other information you think would describe the test assembly to someone outside TerraPower.

Thanks!

Reply:
@albeanth Apparently, this is your test assembly.  Just thinking aloud.